src/main/java/ben/parser/Parser.java

```diff
@@ -0,0 +1,154 @@
+                if (tokens.length < 2) {
+                    throw new BenException("   Key in a value!");
+                }
+
+                // obtain index of task within Ben.tasks.TaskList
+                int index = Integer.parseInt(tokens[1]) - 1;
+
+                return new UnmarkCommand(index);
+            }
+
+            case "todo": {
+                // Todo command
+
+                // empty to-do
+                if (tokens.length < 2) {
+                    throw new BenException("   OOPS!!! The description of a todo cannot be empty.");
+                }
+
+                String description = tokens[1];
+
+                return new TodoCommand(description);
+            }
+
+            case "deadline": {
+                // Deadline command
+
+                // empty deadline
+                if (tokens.length < 2) {
+                    throw new BenException("   OOPS!!! The description of a deadline cannot be empty.");
+                }
+
+                // delimiting string
+                String information = tokens[1];
+                String[] descTokens = information.split(" /by ");
+                String description = descTokens[0];
+                String by = descTokens[1];
+
+                try {
+                    LocalDate deadline = LocalDate.parse(by);
+                    return new DeadlineCommand(description, deadline);
+                } catch (DateTimeParseException e) {
+                    throw new BenException("Invalid deadline format");
+                }
+            }
+
+            case "event": {
+                // Event command
+
+                // empty event
+                if (tokens.length < 2) {
+                    throw new BenException("   OOPS!!! The description of an event cannot be empty.");
+                }
+
+                // delimiting string
+                String information = tokens[1];
+                String[] descTokens = information.split(" /from ");
+                String description = descTokens[0];
+                String dates = descTokens[1];
+                String[] dateTokens = dates.split(" /to ");
+                String startDate = dateTokens[0];
+                String endDate = dateTokens[1];
+
+                try {
+                    LocalDate dateFormattedStartDate = LocalDate.parse(startDate);
+                    LocalDate dateFormattedEndDate = LocalDate.parse(endDate);
+                    return new EventCommand(description, dateFormattedStartDate, dateFormattedEndDate);
+                } catch (DateTimeParseException e) {
+                    throw new BenException("Invalid deadline format");
+                }
+            }
+
+            case "delete": {
+                // Delete command
+
+                // empty field
+                if (tokens.length < 2) {
+                    throw new BenException("   Key in a value");
+                }
+
+                int index = Integer.parseInt(tokens[1]) - 1;
+
```

Comment:
There is some duplicated code for parsing the index of a task (e.g., in the "mark," "unmark," and "delete" cases). Maybe extracting this logic into a separate method can to avoid redundancy? (if it doesn't affect readability)